server/src/daemon/collection.rs

```diff
@@ -0,0 +1,199 @@
+// org.freedesktop.Secret.Collection
+
+use std::{
+    collections::HashMap,
+    sync::{atomic::AtomicBool, Arc, Mutex},
+    time::Duration,
+};
+
+use oo7::{
+    dbus::api::{Properties, Secret, SecretInner},
+    portal::{Item, Keyring},
+};
+use tokio::sync::RwLock;
+use zbus::{interface, zvariant, ObjectServer, SignalContext};
+use zvariant::{ObjectPath, OwnedObjectPath};
+
+use super::{
+    error::ServiceError, prompt::Prompt, service_manager::ServiceManager, Result, Service,
+};
+
+const SECRET_COLLECTION_PREFIX: &str = "/org/freedesktop/secrets.Devel/collection/";
+
+#[derive(Debug)]
+pub struct Collection {
+    keyring: Arc<Keyring>,
+    pub(crate) items: RwLock<Vec<super::item::Item>>,
+    alias: RwLock<String>,
+    label: String,
+    locked: AtomicBool,
+    created: Duration,
+    modified: Duration,
+    manager: Arc<Mutex<ServiceManager>>,
+    path: OwnedObjectPath,
+}
+
+#[interface(name = "org.freedesktop.Secret.Collection")]
+impl Collection {
+    pub async fn delete(
+        &self,
+        #[zbus(signal_context)] ctxt: SignalContext<'_>,
+        #[zbus(object_server)] object_server: &ObjectServer,
+    ) -> Result<Prompt> {
+        let _ = object_server.remove::<Self, _>(&self.path).await;
+        Service::collection_deleted(&ctxt, self.path.as_ref()).await?;
+        Ok(Prompt::default())
+    }
+
+    pub async fn search_items(&self, attributes: HashMap<&str, &str>) -> Result<Vec<Item>> {
+        // TODO currently returns Vec<oo7::portal::Item>, this should be a path
+        self.keyring
+            .search_items(&attributes)
+            .await
+            .map_err(From::from)
+    }
+
+    pub async fn create_item(
+        &self,
+        #[zbus(signal_context)] ctxt: SignalContext<'_>,
+        properties: Properties,
+        secret: SecretInner,
+        replace: bool,
+    ) -> Result<(OwnedObjectPath, Prompt)> {
+        let label = properties.label();
+        let attributes = properties.attributes().unwrap();
+
+        let session = secret.0;
+        let parameters = secret.1;
+        let value = secret.2;
+        let content_type = secret.3;
+
+        // TODO: Use the session to encrypt the data
+        // TODO: figure out what should be done regarding the content type
+
+        let session = self
+            .manager
+            .lock()
+            .unwrap()
+            .session(session.into())
+            .unwrap();
+        let aes_key = session.aes_key();
+
+        let secret = if aes_key.is_none() {
+            Secret::new(Arc::new(session), value, content_type.as_str()) // private function
+        } else {
+            Secret::new_encrypted(
+                Arc::new(session), // expected `Session<'_>`, found `Session`
+                value,
+                content_type.as_str(),
+                &aes_key.unwrap(),
+            )
```

Comment:
is this approach make sense to encrypt (or not) the data using session? @bilelmoussaoui 
one issue:
- `Secret::new_encrypted()` expecting a `Session<'_>` and I'm passing a `Session` :) is adding a lifetime to [Session](https://github.com/bilelmoussaoui/oo7/blob/1419233f39ef1807b7ffd5aa62ce40ab97f0c035/server/src/daemon/session.rs#L14) could fix this mismatched types issue?